File: backend/pkg/models/buyer.go
package models

type Buyer struct {
	Uid          string        `json:"uid,omitempty"`
	Id           string        `json:"id,omitempty"`
	Name         string        `json:"name,omitempty"`
	Age          int           `json:"age,omitempty"`
	Transactions []Transaction `json:"made,omitempty"`
	Date         string        `json:"date,omitempty"`
	DType        []string      `json:"dgraph.type,omitempty"`
}

func NewBuyer(id string, name string, age int, date string) (*Buyer, error) {
	return &Buyer{
		Id:    id,
		Name:  name,
		Age:   age,
		DType: []string{"Buyer"},
		Date:  date,
	}, nil
}

File: transactions/app/src/main/java/com/example/transactions/models/Buyer.kt
package com.example.transactions.models

import com.google.gson.annotations.SerializedName

data class Buyer(
        @SerializedName("id")
        val id: String,
        @SerializedName("name")
        val name: String,
        @SerializedName("age")
        val age: Int,
        @SerializedName("date")
        val date: String
)
